src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -110,55 +92,44 @@ protected static DefaultConfiguration createModuleConfig(Class<?> clazz) {
      * @return {@link Checker} instance based on the given {@link Configuration} instance.
      * @throws Exception if an exception occurs during checker configuration.
      */
-    public final Checker createChecker(Configuration moduleConfig)
+    protected final Checker createChecker(Configuration moduleConfig)
             throws Exception {
-        ModuleCreationOption moduleCreationOption = ModuleCreationOption.IN_CHECKER;
-
         final String moduleName = moduleConfig.getName();
-        if (!ROOT_MODULE_NAME.equals(moduleName)) {
-            try {
-                final Class<?> moduleClass = Class.forName(moduleName);
-                if (ModuleReflectionUtil.isCheckstyleTreeWalkerCheck(moduleClass)
-                        || ModuleReflectionUtil.isTreeWalkerFilterModule(moduleClass)) {
-                    moduleCreationOption = ModuleCreationOption.IN_TREEWALKER;
-                }
-            }
-            catch (ClassNotFoundException ignore) {
-                // ignore exception, assume it is not part of TreeWalker
-            }
+        final Checker checker = new Checker();
+        checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
+
+        if (ROOT_MODULE_NAME.equals(moduleName)) {
+            checker.configure(moduleConfig);
         }
+        else {
+            final Class<?> moduleClass = Class.forName(moduleName);
 
-        return createChecker(moduleConfig, moduleCreationOption);
+            addConfigToChecker(checker, moduleClass, moduleConfig);
+        }
+
+        checker.addListener(getBriefUtLogger());
+        return checker;
     }
 
     /**
-     * Creates {@link Checker} instance based on the given {@link Configuration} instance.
+     * Identifies the way to add the {@code moduleConfig} to the {@code checker} instance.
      *
+     * @param checker {@link Checker} instance.
+     * @param moduleClass {@link Class} of the module involved.
      * @param moduleConfig {@link Configuration} instance.
-     * @param moduleCreationOption {@code IN_TREEWALKER} if the {@code moduleConfig} should be added
-*                                              under {@link TreeWalker}.
-     * @return {@link Checker} instance based on the given {@link Configuration} instance.
-     * @throws Exception if an exception occurs during checker configuration.
+     * @throws Exception if an exception occurs during configuration.
      */
-    public final Checker createChecker(Configuration moduleConfig,
-                                 ModuleCreationOption moduleCreationOption)
-            throws Exception {
-        final Checker checker = new Checker();
-        checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
-
-        if (moduleCreationOption == ModuleCreationOption.IN_TREEWALKER) {
+    protected void addConfigToChecker(Checker checker, Class<?> moduleClass,
+            Configuration moduleConfig) throws Exception {
```

Comment:
This is the method that can be overridden to allow new walkers and why the old enum way had to be removed.